### DISCUSSION TITLE
Help: Need Urgent Help for Configuring OAuth2 in Chamilo

### DISCUSSION THREAD

**Discussion by Moigaston** [chamilo/chamilo-lms#5397] [General]
The OAuth2 plugin in Chamilo is impossible to configure for a non-expert like me. The IT specialist working for my school (in Germany) refuses to configure it because he is not familiar with Chamilo and claims he cannot find the necessary information to do so. Chamilo is virtually unknown in Germany. To my knowledge, only the Languages department at Leipzig University uses it (https://chamilo.spracheninstitut-leipzig.de/). In Germany, they only know Moodle. The survival of Chamilo in my school depends on the configuration of OAuth2, so thank you for any assistance. Otherwise, it will be Moodle (where configuring the Auth2 plugin is extremely easy, even for a non-expert like me). The decision will be taken next week. Question: Is a minimalist configuration possible with just the following information (instead of 32 pieces of information): Client ID, Client Secret, Authorize URL, Scopes, Access Token HTTP Method: POST, Response error key, Response Resource Owner Id key. I asked the question to ChatGPT, and it replied that it should work. What do you think? Thanks for any response.

**Comment by LudiscapeXApi** [comment]
Hello,

For a non-expert, the solution would be to use the CAS protocol, which is configurable on all federated identity systems.

Otherwise, it would be advisable to seek the services of an official provider. From memory, [Beeznest ](https://beeznest.com/nl/contact-opnemen) is very competent in this field and could assist you effectively.

Best regards,

Damien

**Comment by Moigaston** [comment]
I've got it! I did it and it works now!

**Reply by dannymk** [reply]
Can you share how you did it?  I tried it and received nothing but blank pages.  I have configured many applications using CAS, this is the first one that stumped me.

**Reply by dannymk** [reply]
I got CAS working if you need to figure this out.  My authentication server uses many protocols and this is one.  Once I figured the correct options in the Chamilo configuration files and interface it worked well.

**Comment by LudiscapeXApi** [comment]
Bien content pour toi Gaston !

**Comment by lonesomewalker** [comment]
I can assure, not only Spracheninstitut Leipzig (who is by the way our customer) uses Chamilo.
But it has many reasons why Chamilo is not widespread...
